File: java/translate/TokenAuth.java
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class TokenAuth {
        public static void main(String args[]) {
                // basic configuration
                String clientId = "[[client_id]]";
                String clientSecret = "[[client_secret]]";
                String scopes = "[[scopes]]";

                // get the access_token from the token endpoint
                String tokenPayload = "{ \"client_id\": \"" + clientId + "\", \"client_secret\": \"" + clientSecret
                                + "\", \"grant_type\": \"client_credentials\", \"scope\": \"" + scopes + "\"  }";
                String token = "";

                try {
                        HttpClient client = HttpClient.newHttpClient();
                        HttpRequest request = HttpRequest.newBuilder()
                                        .uri(URI.create("https://api.intelligent-api.com/v1/token"))
                                        .header("Content-Type", "application/json")
                                        .POST(HttpRequest.BodyPublishers.ofString(
                                                        tokenPayload))
                                        .build();

                        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

                        String searchKey = "access_token";
                        String tokenResponse = response.body();
                        int keyIndex = tokenResponse.indexOf(searchKey);
                        int valueStart = tokenResponse.indexOf("\"", keyIndex + searchKey.length()) + 3;
                        int valueEnd = tokenResponse.indexOf("\"", valueStart);
                        token = tokenResponse.substring(valueStart, valueEnd);
                } catch (Exception e) {
                        e.printStackTrace();
                        return;
                }

                // use the access_token as a bearer token
                String authorization = "Bearer ".concat(token);

                // populate endpoint parameters
                String textBody = "[[text_body]]";
                String sourceLanguage = "[[source_language]]";
                String targetLanguage = "[[target_language]]";
                String userAgent = "Intelligent API Sample Java Code";

                String jsonPayload = "{ \"text\": \"" + textBody + "\", \"sourceLanguage\": \"" + sourceLanguage
                                + "\", \"targetLanguage\": \"" + targetLanguage + "\" }";

                // invoke the API endpoint
                try {
                        HttpClient client = HttpClient.newHttpClient();
                        HttpRequest request = HttpRequest.newBuilder()
                                        .uri(URI.create("https://api.intelligent-api.com/v1/language/translate"))
                                        .header("Authorization",
                                                        authorization)
                                        .header("User-Agent",
                                                        userAgent)
                                        .header("Content-Type", "application/json")
                                        .POST(HttpRequest.BodyPublishers.ofString(
                                                        jsonPayload))
                                        .build();

                        HttpResponse<String> response = client.send(request,
                                        HttpResponse.BodyHandlers.ofString());

                        System.out.println("Response Code: " + response.statusCode());
                        System.out.println("Response Body: " + response.body());
                } catch (Exception e) {
                        e.printStackTrace();
                }
        }
}

File: python/translate/token-auth.py
import base64
import http.client
import json

# basic configuration
client_id = "[[client_id]]"
client_secret = "[[client_secret]]"
scopes = "[[scopes]]"

# get the access_token from the token endpoint
token_data = {
    "client_id": client_id,
    "client_secret": client_secret,
    "grant_type": "client_credentials",
    "scope": scopes
}

token_json_data = json.dumps(token_data)

token_host = "api.intelligent-api.com"
token_path = "/v1/token"

token_connection = http.client.HTTPSConnection(token_host)

token_headers = {
    "Content-Type": "application/json"
}

token_connection.request("POST", token_path, body=token_json_data, headers=token_headers)
token_response = token_connection.getresponse()
token_response_data = token_response.read().decode("utf-8")

token_json_response = json.loads(token_response_data)
token = token_json_response.get("access_token") 

# use the access_token as a bearer token
authorization = "Bearer " + token

# populate endpoint parameters
text = "[[text_body]]"
source_language = "[[source_language]]"
target_language = "[[target_language]]"
user_agent = "Intelligent API Sample Python Code"

data = {
    "text": text,
    "sourceLanguage": source_language,
    "targetLanguage": target_language
}

json_data = json.dumps(data)

# invoke the API endpoint
host = "api.intelligent-api.com"
path = "/v1/language/translate"

connection = http.client.HTTPSConnection(host)

headers = {
    "Authorization": authorization,
    "Content-Type": "application/json",
    "User-Agent": user_agent
}

connection.request("POST", path, body=json_data, headers=headers)
response = connection.getresponse()
response_data = response.read().decode("utf-8")

print("Status Code:", response.status)
print("Response:", response_data)

connection.close()